core/CacheId.php

```diff
@@ -26,4 +26,46 @@ public static function pluginAware($cacheId)
 
         return $cacheId;
     }
+
+    public static function siteAware($cacheId, array $idSites = null)
+    {
+        if ($idSites === null) {
+            $idSites = self::getIdSiteList('idSite');
+            $cacheId .= self::idSiteListCacheKey($idSites);
+
+            $idSites = self::getIdSiteList('idSites');
+            $cacheId .= self::idSiteListCacheKey($idSites);
+        } else {
+            $cacheId .= self::idSiteListCacheKey($idSites);
+        }
+
+        return $cacheId;
+    }
+
+    private static function getIdSiteList($queryParamName)
+    {
+        $idSiteParam = Common::getRequestVar($queryParamName, false);
```

Comment:
@benaka I just had a look how I handled this cacheId implementation just recently and found https://github.com/matomo-org/matomo/blob/3.2.0/core/Columns/MetricsList.php#L97-L108 . However, it does not take into consideration `idSites` but the different `$_GET` and `$_POST` values. I wonder, to be safe, if we should mix the logic of that implementation for the cache and this implementation. 

Specifically, `MetricsList` is missing the `idSites` parameter, and here we could add to also check for `$_GET and $_POST` and maybe also `idsite` in lower case. The lower case `idsite` is important to recognize as well when someone is using `CacheId::siteAware()` during tracking. 

Looking at $_GET and $_POST and adding both values to the cache key could prevent some random security issues if someone passes different values in $_GET and $_POST and for some reason some plugin only looks at $_GET (mainly 3rd party plugins I would say as we always use `Common::getRequestVar` and in most cases do not set a custom `$requestArrayToUse`).